src/client/app/component/HomePageHeader/index.js

```diff
@@ -35,21 +43,39 @@ class HomePageHeader extends React.Component {
   }
 
   render() {
-    const btnText = this.state.isLoggedIn ? 'Log Out' : 'Log In';
+    const userText = localStorage.getItem('Username') === null ?
+      'Log Out' : localStorage.getItem('Username');
+    const btnText = this.state.isLoggedIn ? userText : 'Log In';
+    let button = null;
+    const menu = (
+      <Menu onClick={this.onClickHeaderLogBtn.bind(this)}>
+        <Menu.Item key="1">Log Out</Menu.Item>
+      </Menu>);
+
+    if (btnText !== 'Log In') {
+      button =
+        <Dropdown overlay={menu} size="small">
+          <a className="ant-dropdown-link" >
+            {btnText}
+          </a>
+        </Dropdown>;
+    } else {
+      button = <button onClick={this.onClickHeaderLogBtn.bind(this)}>
```

Comment:
separate the two functionalities. (maybe you can solve it something like `this.props.history.push`)